apps/zk-explorer/app/page.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+
+import {
+  Button,
+  Input,
+  Card,
+  Typography,
+} from '@webb-tools/webb-ui-components';
+import { ArrowRightUp } from '@webb-tools/icons';
+import { useState } from 'react';
+import { Link } from '@webb-tools/webb-ui-components/components/Link';
+import { validateGithubUrl } from '../utils';
+
+export default function Index() {
+  const [githubUrl, setGithubUrl] = useState('');
+  const [isValidGithubUrl, setIsValidGithubUrl] = useState(false);
+  const [isGithubUrlDisplayingError, setIsGithubUrlDisplayingError] =
+    useState(false);
+
+  return (
+    <main className="flex flex-col gap-6 pt-6">
+      <Typography variant="h4" fw="bold">
+        Upload Project
+      </Typography>
+
+      <div className="flex flex-col sm:flex-row gap-6 items-start">
+        <Card className="max-w-[780px]">
+          <Typography variant="h5">1. GitHub Repository URL:</Typography>
+
+          <Input
```

Comment:
Consider creating a new component for this Input and moving all the hooks into it. We should handle client actions in components rather making the whole page client-side